src/psyclone/dynamo0p3.py

```diff
@@ -9110,62 +9147,115 @@ def ref_name(self, function_space=None):
                     break
             if not found:
                 raise GenerationError(
-                    "DynKernelArgument:ref_name(fs). The supplied function "
+                    "DynKernelArgument.ref_name(fs): The supplied function "
                     "space (fs='{0}') is not one of the function spaces "
-                    "associated with this argument (fss='{1}')".format(
+                    "associated with this argument (fss={1}).".format(
                         function_space.orig_name,
                         self.function_space_names))
-        if self._type in LFRicArgDescriptor.VALID_FIELD_NAMES:
+        if self.is_field():
             return "vspace"
-        if self.is_operator:
+        if self.is_operator():
```

Comment:
For consistency with `is_scalar()` which is not a property but a method because it is set that way in the base `KernelArgument` class. As I said in a comment below, I am happy to do method to property from ground up to the APIs that use it.

Reply:
P.S. There were not so many instances in the code so I changed the method into a property (starting with the abstract property in the `KernelArgument` class).

Reply:
That's great - thanks Iva. I think it's much better now.